examples/misc/agg_buffer_to_array.py

```diff
@@ -7,9 +7,12 @@
 """
 import matplotlib.pyplot as plt
 import numpy as np
+from matplotlib.figure import Figure
 
-# make an agg figure
-fig, ax = plt.subplots()
+# Create a figure that pyplot does not know about.  If no canvas is
+# specified, will default to `matplotlib.backends.FigureCanvas`.
```

Comment:
no, it will default to FigureCanvasBase, which has no renderer attribute.

what you're looking for is either manually attaching an agg canvas `FigureCanvasAgg(figure)` or using savefig, which will insert the correct canvas for the required output format `fig.savefig(buf, format="rgba")` (the latter makes a copy, but in the example here np.array also makes a copy so heh).

Reply:
sigh, I fixed one sleep deprived error with another sleep deprived error....